python/tk_houdini_alembicnode/handler.py

```diff
@@ -0,0 +1,521 @@
+
+    ############################################################################
+    # Class data
+
+    HOU_ROP_ALEMBIC_TYPE = "alembic"
+    """Houdini type for alembic rops."""
+
+    HOU_SOP_ALEMBIC_TYPE = "rop_alembic"  
+    """Houdini type for alembic sops."""
+    # this is correct. the houdini internal rop_alembic is a sop.
+
+    NODE_OUTPUT_PATH_PARM = "filename"
+    """The name of the output path parameter on the node."""
+
+    TK_ALEMBIC_NODE_TYPE = "sgtk_alembic"
+    """The class of node as defined in Houdini for the Alembic nodes."""
+
+    TK_CONFIG_PARM = "alembic_config"
+    """The name of the parameter that stores the toolkit node configuration."""
+
+    TK_CONFIG_NAME_KEY = "tk_config_name"
+    """The name of the key in the user data that stores the config name."""
+
+    ############################################################################
+    # Class methods
+
+    @classmethod
+    def convert_back_to_tk_alembic_nodes(cls, app):
+        """Convert Alembic nodes back to Toolkit Alembic nodes.
+
+        :param app: The calling Toolkit Application
+
+        Note: only converts nodes that had previously been Toolkit Alembic
+        nodes.
+
+        """
+
+        # get all rop/sop alembic nodes in the session
+        alembic_nodes = []
+        alembic_nodes.extend(hou.nodeType(hou.sopNodeTypeCategory(),
+            cls.HOU_SOP_ALEMBIC_TYPE).instances())
+        alembic_nodes.extend(hou.nodeType(hou.ropNodeTypeCategory(),
+            cls.HOU_ROP_ALEMBIC_TYPE).instances())
+
+        if not alembic_nodes:
+            app.log_debug("No Alembic Nodes found for conversion.")
+            return
+
+        # the tk node type we'll be converting to
+        node_type = TkAlembicNodeHandler.TK_ALEMBIC_NODE_TYPE
+
+        # iterate over all the alembic nodes and attempt to convert them
+        for alembic_node in alembic_nodes:
+
+            # get the user data dictionary stored on the node
+            user_dict = alembic_node.userDataDict()
+
+            # get the config data from the dictionary
+            tk_config_name = user_dict.get(cls.TK_CONFIG_NAME_KEY)
+
+            if not tk_config_name:
+                app.log_warning(
+                    "Almbic node '%s' does not have tk config name. "
+                    "Can't convert to Tk Alembic node. Continuing." %
+                    (alembic_node.name(),)
+                )
+                continue
+
+            # create a new, Toolkit Alembic node:
+            tk_alembic_node = alembic_node.parent().createNode(node_type)
+
+            # find the index of the stored name on the new tk alembic node
+            # and set that item in the menu.
+            try:
+                config_parm = tk_alembic_node.parm(
+                    TkAlembicNodeHandler.TK_CONFIG_PARM)
+                index = config_parm.menuLabels().index(tk_config_name)
+                config_parm.set(index)
+            except ValueError:
+                pass
```

Comment:
ValueError will be raised by index() when there is no configuration of that name we want to set it back to. I agree that there should be a warning because it means the configuration will be set to a different value and this could result in different output paths.